query/aggregate_test.go

```diff
@@ -0,0 +1,451 @@
+// Copyright 2015 Square Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package query
+
+import (
+	"github.com/square/metrics/api"
+
+	"math"
+	"testing"
+)
+
+var (
```

Comment:
AFAIK, all the test suites in the file scope in this file are independent of each other, right? In that case, let's move them into each of the functions instead - it's a bit hard to read because i need to jump between the test case and the test function they're being used in.